cpu/native/eui_provider/eui_provider.c

```diff
@@ -0,0 +1,59 @@
+/**
+ * Native CPU EUI provider
+ *
+ * Copyright (C) 2020 Benjamin Valentin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup cpu_native
+ * @{
+ * @file
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "net/l2util.h"
+
+/* list of user supplied EUI-64s */
+struct _native_eui64_list;
+static struct _native_eui64_list {
+    eui64_t addr;
+    struct _native_eui64_list *prev;
+} *_eui64_head;
+
+/* parse EUI-64 from command line */
+void native_add_eui64(const char *s)
+{
+    struct _native_eui64_list *e = malloc(sizeof(*_eui64_head));
```

Comment:
```suggestion
    struct _native_eui64_list *e = real_malloc(sizeof(*_eui64_head));
```
To use the host's malloc instead of the native instance's malloc?

Reply:
What's the advantage here?

Reply:
Not using up the pool if the `native` instance uses e.g. `tlsf` for `malloc`.

Reply:
But 

```C
_native_syscall_enter();
struct _native_eui64_list *e = real_malloc(sizeof(*_eui64_head));
_native_syscall_leave();
```

is objectively more ugly.

Reply:
Why add the `_native_syscall` thingies around?

Reply:
I've never seen that and not using them and the syscalls directly within e.g. `socket_zep`, so I am a bit confused.

Reply:
I've noticed them around all `real_` function calls that might result in a syscall.
I don't know why they are needed, but there probably is a reason.

Reply:
Ok, but can we just try for now to just use it without the surrounding calls and see how it goes?

Reply:
`malloc` doesn't always do a system call for each allocation, typically it allocates a pool and only if that is exhausted it will do a system call to request more memory.

Reply:
So we don't have to surround `real_malloc()`?

Reply:
Well just because most calls to `real_malloc()` won't result in a syscall doesn't mean that all calls won't :wink: